### PR TITLE
Move Intel FPGA to Python testing infrastructure

### PR DESCRIPTION
**Pull request by definelicht**
Re-use the testing infrastructure for Xilinx for Intel FPGA tests as well.

Moved all FPGA-specific tests under the `tests/fpga` subfolder.

Adjust some samples to apply `FPGATransformSDFG` instead of using the command-line interface.

### CODE REVIEW

**Review by TizianoDeMatteis — APPROVED** [review]
Super useful! Thanks!

**Comment by tbennun** [comment]
@definelicht you should let me review such PRs too...

File: .github/workflows/python-package.yml [hunk 1]
```diff
@@ -121,24 +121,23 @@ jobs:
...
-    - name: Run Xilinx tests
+    - name: Run Intel FPGA tests
       run: |
         export NOSTATUSBAR=1
         export DACE_testing_single_cache=0
         export COVERAGE_RCFILE=`pwd`/.coveragerc
         export PYTHON_BINARY="coverage run --source=dace --parallel-mode"
         . /opt/setupenv
-        python tests/xilinx_test.py
+        python tests/fpga/intel_fpga_test.py
```
Comment (tbennun) [inline .github/workflows/python-package.yml:131]:
```suggestion
        $PYTHON_BINARY tests/fpga/intel_fpga_test.py
```

File: .github/workflows/python-package.yml [hunk 2]
```diff
@@ -121,24 +121,23 @@ jobs:
...
       run: |
         export NOSTATUSBAR=1
-        export DACE_testing_single_cache=1
+        export DACE_testing_single_cache=0
         export COVERAGE_RCFILE=`pwd`/.coveragerc
         export PYTHON_BINARY="coverage run --source=dace --parallel-mode"
         . /opt/setupenv
-        export DACE_compiler_default_data_types=C
-        ./tests/intel_fpga_test.sh
+        python tests/fpga/xilinx_test.py
```
Comment (tbennun) [inline .github/workflows/python-package.yml:140]:
```suggestion
        $PYTHON_BINARY tests/fpga/xilinx_test.py
```

**Review by tbennun — COMMENTED** [review]
Requesting changes retroactively 🐍

**Comment by definelicht** [comment]
I tried to restore the branch and push new changes, but it doesn't seem to be possible to merge a second time -- and now I can't delete it either? 🤔 Should I force-delete the branch from the CLI?

**Pull request merged**